src/node.cc

```diff
@@ -4296,12 +4296,18 @@ Environment* CreateEnvironment(IsolateData* isolate_data,
   HandleScope handle_scope(isolate);
   Context::Scope context_scope(context);
   auto env = new Environment(isolate_data, context);
+  CHECK_EQ(0, uv_key_create(&thread_local_env));
+  uv_key_set(&thread_local_env, env);
```

Comment:
Maybe move this to `Environment::Start()`?  That way you don't duplicate it here and in `node::Start()`.

Reply:
Sounds good, I'll take a look. Thanks